### PR TITLE
Fixed Error propagation

### PR DESCRIPTION
**Pull request by mtrabelsi**
before change any try/catch surrounding the .**attach** function is useless.
with this PR we forced throwing error -when something is not OK-, is is a must for error handling in userland.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -256,8 +256,7 @@ const attach = async (trytes, root, depth = 6, mwm = 14) => {
         )
         return objs
     } catch (e) {
-        console.error('failed to attach message:', '\n', e)
-        return e
+       	throw `failed to attach message: ${e}`
```
Comment (mtrabelsi) [inline src/index.js:259]: don't know why github changed the tabs

Comment (mtrabelsi) [reply src/index.js:259]: double checking this, looks fine on VIM & VS

**Comment by sobolev-alexey** [comment]
could you please also re-build `mam.web.js` with the same fix?

**Review by chrisdukakis — APPROVED** [review]
LGTM

**Review by sobolev-alexey — APPROVED** [review]

**Pull request merged**